### PR TITLE
Added user rebinding on reconnect in LDAP

### PR DESCRIPTION
**Pull request by aidenfoxx**
The user binding would be lost on reconnect. Added code to rebind user when this happens.

### CODE REVIEW

File: api/src/auth/drivers/ldap.ts [hunk 1]
```diff
@@ -56,26 +56,40 @@ export class LDAPAuthDriver extends AuthDriver {
...
-						reject(new InvalidConfigException('Invalid bind user', { provider: additionalConfig.provider }));
-					} else {
-						reject(error);
+			client.on('connect', () => {
+				client.bind(bindDn, bindPassword, (err: Error | null) => {
+					if (err) {
+						const error = handleError(err);
+
+						if (!isClientPending) {
+							logger.error(err);
```
Comment (aidenfoxx) [inline api/src/auth/drivers/ldap.ts:81]:
```suggestion
							logger.error(error);
```

**Comment by dorianim** [comment]
Seems to work, but the first logon attempt still fails with an unexpected error.

**Comment by aidenfoxx** [comment]
@dorianim I can't reproduce the issue with the current code. I don't think it will fix things, but it would be good to know what error you get when attempting to log in the first time after 30 minutes?

**Comment by dorianim** [comment]
@aidenfoxx ok, I'll test :)

**Comment by dorianim** [comment]
I tested it again, and it just keeps loading forever after I click login. The Requests are sent to the backend, but there is no response.

**Comment by aidenfoxx** [comment]
ARG! Really sucks that I can't reproduce this locally. Need to figure out what controls the bind session length so I can debug it properly 😞

**Comment by dorianim** [comment]
And now it just works ...
Really strange

**Comment by dorianim** [comment]
>  Really sucks that I can't reproduce this locally

If you want, I can give you access to my samba server. It is an isolated test machine anyway. Might help for testing. You can just send me a public WireGuard key to  mail@dorian.im, and I will give you access if you want.

**Comment by harish0sim** [comment]
Please also dump redacted request and response header and payload. Might help in reproduction.

**Comment by aidenfoxx** [comment]
@dorianim That should fix it. Now we do a healthcheck whenever we need to call LDAP, and re-bind the user if required.

**Comment by dorianim** [comment]
Nice! I will test as soon as possible :D

**Comment by dorianim** [comment]
It looks like it works :D

**Comment by aidenfoxx** [comment]
@dorianim Perfect! And I'm happy enough with the solution since it feels we get the best of both worlds. 

Thanks for all your help in debugging this. Hopefully this solves any outstanding issues. 🤞

**Review by rijkvanzanten — APPROVED** [review]

**Pull request merged**